tools/analyzer_plugin/test/integration/diagnostics/callback_ref_test.dart

```diff
@@ -0,0 +1,281 @@
+  },
+  UiFactoryConfig(displayName: 'UsesCallbackRef'),
+);
+''';
+
+  static const usageSourceWithinFnComponentFixedTheRest = '''
+import 'package:over_react/over_react.dart';
+
+final UsesCallbackRef = uiFunction<UiProps>(
+  (props) {
+    ${CallbackRefDiagnosticTest.refVarDeclarationsFixedTheRest}
+
+    ${CallbackRefDiagnosticTest.refUsageFixedTheRest}
+
+    ${CallbackRefDiagnosticTest.renderReturnFixedTheRest}
+  },
+  UiFactoryConfig(displayName: 'UsesCallbackRef'),
+);
+''';
+
+  Future<void> test_someCustomRefError() async {
+    final source = newSource('test.dart', usageSourceWithinFnComponent);
+    await expectSingleErrorAt(createSelection(source, CallbackRefDiagnosticTest.selectionToFixSome));
+  }
+
+  Future<void> test_someCustomRefErrorFix() async {
+    var source = newSource('test.dart', usageSourceWithinFnComponent);
+    final errorFix = await expectSingleErrorFix(createSelection(source, CallbackRefDiagnosticTest.selectionToFixSome));
+    expect(errorFix.fixes.single.change.selection, isNull);
+    source = applyErrorFixes(errorFix, source);
+    expect(source.contents.data, usageSourceWithinFnComponentFixedSome);
+  }
+
+  Future<void> test_anotherCustomRefError() async {
+    final source = newSource('test.dart', usageSourceWithinFnComponentFixedSome);
+    await expectSingleErrorAt(createSelection(source, CallbackRefDiagnosticTest.selectionToFixAnother));
+  }
+
+  Future<void> test_anotherCustomRefErrorFix() async {
+    var source = newSource('test.dart', usageSourceWithinFnComponentFixedSome);
+    final errorFix =
+        await expectSingleErrorFix(createSelection(source, CallbackRefDiagnosticTest.selectionToFixAnother));
+    expect(errorFix.fixes.single.change.selection, isNull);
+    source = applyErrorFixes(errorFix, source);
+    expect(source.contents.data, usageSourceWithinFnComponentFixedTheRest);
+  }
+}
+
+@reflectiveTest
+class CallbackRefDiagnosticClassComponentTest extends CallbackRefDiagnosticTest {
+  static const usageSourceWithinClassComponent = '''
+import 'package:over_react/over_react.dart';
+
+part 'test.over_react.g.dart';
+
+UiFactory<UsesCallbackRefProps> UsesCallbackRef = castUiFactory(_\$UsesCallbackRef); // ignore: undefined_identifier
+
+mixin UsesCallbackRefProps on UiProps {}
+
+class UsesCallbackRefComponent extends UiComponent2<UsesCallbackRefProps> {
+  ${CallbackRefDiagnosticTest.refVarDeclarations}
+
+  @override
+  render() {
+    ${CallbackRefDiagnosticTest.renderReturn}
+  }
+
+  ${CallbackRefDiagnosticTest.refUsage}
+}
+''';
+
+  static const usageSourceWithinClassComponentFixedSome = '''
+import 'package:over_react/over_react.dart';
+
+part 'test.over_react.g.dart';
+
+UiFactory<UsesCallbackRefProps> UsesCallbackRef = castUiFactory(_\$UsesCallbackRef); // ignore: undefined_identifier
+
+mixin UsesCallbackRefProps on UiProps {}
+
+class UsesCallbackRefComponent extends UiComponent2<UsesCallbackRefProps> {
+  ${CallbackRefDiagnosticTest.refVarDeclarationsFixedSome}
```

Comment:
Isn't this using useRef instead of createRef?

It might be nice to inline the ref declarations so it's harder to miss this kind of mismatch when reading the code

Reply:
The class component tests actually weren't running at all...

Reply:
And they also weren't working in the analyzer jam base branch.

Enabled the tests, and then fixed the diagnostic for class components in https://github.com/Workiva/over_react/pull/758/commits/d3b8e77cb96c4bd71624f1e42768d273bbc57e6e